client/src/entrypoints/admin/page-chooser.js

```diff
@@ -42,6 +42,14 @@ function createPageChooser(id, openAtParentId, options) {
 
       state = newState;
     },
+    getTextLabel: (opts) => {
+      if (!state) return null;
+      const result = state.adminTitle;
+      if (opts && opts.maxLength && result.length > opts.maxLength) {
```

Comment:
Would it be possible to move this truncation logic into the caller so it doesn't need to be repeated?

Reply:
I think there are a couple of advantages in doing the truncation at the child level:

- it means we can have block-specific behaviour for the truncation - e.g. ListBlock will stop adding items at the point where it can't fit them in full, rather than cutting them off mid-label
- it means that we can return as soon as we reach maxLength, and save some redundant work calling getTextLabel on later child blocks that are just going to get cut off